### PR TITLE
CAPT-271 - CurrentClaim model to handle multiple claims in combined journey

### PR DESCRIPTION
**Pull request by kenfodder**

### CODE REVIEW

**Review by vassyz — APPROVED** [review]
Looks good to me.

**Review by nigel-lowry — APPROVED** [review]

File: app/models/levelling_up_premium_payments/eligibility.rb [hunk 1]
```diff
@@ -6,12 +6,60 @@ class Eligibility < ApplicationRecord
...
+
+    enum eligible_itt_subject: {
+      chemistry: 0,
+      foreign_languages: 1,
+      mathematics: 2,
+      physics: 3,
+      none_of_the_above: 4
+    }, _prefix: :itt_subject
+
+    enum itt_academic_year: {
```
[the remark this answers is not in the document]
Comment (kenfodder) [reply app/models/levelling_up_premium_payments/eligibility.rb:46]: @vassyz On master? I can't find update you mentioned. Sorry if I'm being stupid.

File: app/controllers/reminders_controller.rb [hunk 2]
```diff
@@ -98,7 +98,11 @@ def default_reminder
   def current_claim
     return @current_claim if defined?(@current_claim)
 
-    @current_claim = Claim.find_by_id(session[:claim_id])
+    claims = Claim.where(id: session[:claim_id])
+    cc = CurrentClaim.new(claims: claims)
+
+    # PLACEHOLDER until reminders are setup for the combined journey
```
Comment (slorek) [inline app/controllers/reminders_controller.rb:104]:
```suggestion
    # TODO: PLACEHOLDER until reminders are setup for the combined journey
```

File: app/models/current_claim.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    if [:attributes=, :save, :save!, :update, :update!, :reset_dependent_answers].include?(method_name)
+      claims.each do |c|
+        result = c.send(method_name, *args, &block) unless c == main_claim
+        Rails.logger.info(result.inspect)
+      end
+    end
+
+    result = main_claim.send(method_name, *args, &block)
+    Rails.logger.info(result.inspect)
+    Rails.logger.info("======<END>==============================")
```
[the remark this answers is not in the document]
Comment (kenfodder) [reply app/models/current_claim.rb]: Removed the logging, not needed

File: app/controllers/submissions_controller.rb [hunk 4]
```diff
@@ -3,8 +3,8 @@ class SubmissionsController < BasePublicController
 
   def create
     if current_claim.submit!
-      ClaimMailer.submitted(current_claim).deliver_later
-      ClaimVerifierJob.perform_later(current_claim)
+      ClaimMailer.submitted(current_claim.main_claim).deliver_later
+      ClaimVerifierJob.perform_later(current_claim.main_claim)
```
[the remark this answers is not in the document]
Comment (kenfodder) [reply app/controllers/submissions_controller.rb:8]: TODO comment added.

File: app/models/levelling_up_premium_payments/eligibility.rb [hunk 5]
```diff
@@ -6,12 +6,60 @@ class Eligibility < ApplicationRecord
...
     def policy
       LevellingUpPremiumPayments
     end
 
     # maintains interface
     def ineligible?
     end
+
+    # maintains interface
+    def reset_dependent_answers
```
[the remark this answers is not in the document]
Comment (kenfodder) [reply app/models/levelling_up_premium_payments/eligibility.rb:62]: Yes, TODO comment added.

**Review by slorek — APPROVED** [review]

**Pull request merged**